ui/pages/confirmations/hooks/alerts/transactions/useMultipleApprovalsAlerts.ts

```diff
@@ -0,0 +1,164 @@
+      }
+
+      const { name, amountOrTokenId, tokenAddress } = parseResult;
+
+      // Skip revocations (zero amounts or setApprovalForAll with false)
+      if (amountOrTokenId?.isZero() || parseResult.isRevokeAll) {
+        return;
+      }
+
+      let actualTokenAddress: Hex;
+      let spenderAddress: Hex | undefined;
+
+      switch (name) {
+        case 'approve':
+          if (tokenAddress) {
+            // Permit2 approve
+            actualTokenAddress = tokenAddress;
+            spenderAddress = to;
+          } else {
+            // Regular ERC20 approve
+            actualTokenAddress = to;
+          }
+          break;
+        case 'increaseAllowance':
+        case 'setApprovalForAll':
+          actualTokenAddress = to;
+          break;
+        default:
+          return;
+      }
+
+      const isNFT = name === 'setApprovalForAll';
+
+      approvalsList.push({
+        tokenAddress: actualTokenAddress,
+        spenderAddress,
+        amount: amountOrTokenId,
+        isNFT,
+      });
+    });
+
+    return approvalsList;
+  }, [isBatchTransaction, nestedTransactions]);
+
+  const tokenOutflows = useMemo((): TokenOutflow[] => {
+    if (!simulationDataArray) {
+      return [];
+    }
+
+    return simulationDataArray
+      .filter((change) => change.isDecrease)
+      .map((change) => ({
+        tokenAddress: change.address.toLowerCase() as Hex,
+        amount: new BigNumber(change.difference, 16),
+        isDecrease: change.isDecrease,
+      }));
+  }, [simulationDataArray]);
+
+  const unusedApprovals = useMemo(() => {
+    if (approvals.length === 0) {
+      return [];
+    }
+
+    return approvals.filter((approval) => {
+      if (approval.isNFT) {
+        const hasNFTOutflows = tokenOutflows.some(
+          (outflow) =>
+            outflow.tokenAddress.toLowerCase() ===
+            approval.tokenAddress.toLowerCase(),
+        );
+        return !hasNFTOutflows;
+      }
+
+      const tokenOutflow = tokenOutflows.find(
+        (outflow) =>
+          outflow.tokenAddress.toLowerCase() ===
+          approval.tokenAddress.toLowerCase(),
+      );
+
+      if (!tokenOutflow) {
+        return true;
+      }
+
+      // If there's an outflow but the approval amount is much larger than needed,
+      // we might still want to flag it, but for now we'll only flag completely unused approvals
+      return false;
+    });
+  }, [approvals, tokenOutflows]);
+
+  const shouldShowAlert = unusedApprovals.length > 0;
+
+  const hasUnusedNFTApprovals = unusedApprovals.some(
+    (approval) => approval.isNFT,
+  );
+
+  return useMemo(() => {
+    if (!shouldShowAlert) {
+      return [];
+    }
+
+    return [
+      {
+        field: RowAlertKey.EstimatedApprovalChanges,
+        isBlocking: false,
+        key: 'multipleApprovals',
+        reason: t('alertReasonMultipleApprovals'),
+        content: t('alertContentMultipleApprovals', [
+          hasUnusedNFTApprovals ? t('nfts') : t('tokens'),
```

Comment:
Is this distinction worth the extra code? Could we just say tokens in both cases?

Reply:
If this is a user facing copy, we can say "assets" to refer to either NFTs or tokens